tests/testapps/CTCatalog/CTCatalog/Tests/CTCFontTestViewController.mm

```diff
@@ -0,0 +1,380 @@
+
+    CTFontRef myCFFont = CTFontCreateWithName(CTFontCopyFullName(_font), CTFontGetSize(_font), NULL);
+
+    // Make dictionary for attributed string with font, color, and alignment
+    NSDictionary* attributesDict = [NSDictionary dictionaryWithObjectsAndKeys:(__bridge id)myCFFont,
+                                                                              (id)kCTFontAttributeName,
+                                                                              color.CGColor,
+                                                                              (id)kCTForegroundColorAttributeName,
+                                                                              (__bridge id)paragraphStyle,
+                                                                              (id)kCTParagraphStyleAttributeName,
+                                                                              nil];
+
+    NSString* text = @"jackdaws love my big sphinx of quartz. JACKDAWS LOVE MY BIG SPHINX OF QUARTZ.";
+    CFAttributedStringRef attrString =
+        CFAttributedStringCreate(kCFAllocatorDefault, (__bridge CFStringRef)text, (__bridge CFDictionaryRef)attributesDict);
+
+    CTFramesetterRef framesetter = CTFramesetterCreateWithAttributedString(attrString);
+    CFRelease(attrString);
+
+    // Creates frame for framesetter with current attributed string
+    CTFrameRef frame = CTFramesetterCreateFrame(framesetter, CFRangeMake(0, 0), path, NULL);
+
+    // Draws the text in the frame
+    CTFrameDraw(frame, context);
+
+    // Creates outline
+    CGContextSetLineWidth(context, 2.0);
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    CGContextSetStrokeColorWithColor(context, color.CGColor);
+    CGContextMoveToPoint(context, 0, 0);
+    CGContextAddRect(context, rect);
+    CGContextStrokePath(context);
+
+    CGColorSpaceRelease(colorspace);
+}
+@end
+
+@implementation CTCFontTestViewController {
+    UIPickerView* _fontPicker;
+    CTFontRef _font;
+    NSArray* _fonts;
+
+    UIPickerView* _nameKeyPicker;
+    CFStringRef _nameKey;
+    NSArray* _nameKeys;
+}
+
+- (instancetype)init {
+    if (self = [super init]) {
+        // This inherits from table view so we can't inherit from base, meaning we have to set our title and background here
+        self.title = [[NSStringFromClass([self class]) stringByReplacingOccurrencesOfString:@"CTC" withString:@""]
+            stringByReplacingOccurrencesOfString:@"ViewController"
+                                      withString:@""];
+
+        self.tableView.allowsSelection = NO;
+        self.view.backgroundColor = [UIColor whiteColor];
+
+        // Query all available fonts and add them to array by name
+        NSMutableArray* fonts = [NSMutableArray new];
+        for (NSString* familyName in [UIFont familyNames]) {
+            for (NSString* fontName in [UIFont fontNamesForFamilyName:familyName]) {
+                [fonts addObject:fontName];
+            }
```

Comment:
you can replace this loop with fonts addObjectsFromArray:[UIFont fontNamesForFamilyName:familyName